core/domain/event_services_test.py

```diff
@@ -71,3 +215,62 @@ def test_events_go_into_the_events_queue(self):
         self.assertEqual(
             self.count_jobs_in_taskqueue(taskqueue_services.QUEUE_NAME_EVENTS),
             0)
+
+
+class EventHandlerNameTests(test_utils.GenericTestBase):
+
+    def _get_all_python_files(self):
```

Comment:
Hmm this seems a bit too heavy-handed, I wonder if there's someway we can fake the python files in the code-base and check against those instead?

Reply:
This is the way @seanlip suggested for value generators and visualisations. We need to ensure that there is not an event handler with same name in the codebase.

Reply:
It's fine to keep this structure, but it'd be nicer if `_get_all_python_files` simlpy returned a list of strings in the test file.

Reply:
We need to import the files as modules too to check if they are subclasses of BaseEventHandler or not. What do you suggest I do?

Reply:
Hmm I can't think of anything off the top of my head, and I don't feel super strongly (just felt it would be healthier).

You can ignore this thread!